TreeDB/docs/evidence/vector-partition-local-system-qualification-4019/m2-95c60cbe/run_m2.py

```
@@ -0,0 +1,213 @@
+def stop_nodes(ready: list[dict[str, object]], processes: list[subprocess.Popen[bytes]]) -> None:
+    if not ready:
+        for process in processes:
+            if process.poll() is None:
+                os.killpg(process.pid, signal.SIGTERM)
+    for value in ready:
+        pid = value.get("pid")
+        if isinstance(pid, int):
+            try:
+                os.kill(pid, signal.SIGTERM)
+            except ProcessLookupError:
+                pass
```

Ruff (0.16.1)

[warning] Use `contextlib.suppress(ProcessLookupError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(ProcessLookupError): ...`

(SIM105)